Beta push for 1.0.16 (many fix and enhancement see details)

hthetiot opened the pull request:
It's time to deal with push requests and opened issues.

I will try to fix as mush bug and do small enhancements under review of @omochi and authority of @EricDavies.

- Add getSourceList and getAudioSourceList #80
- easyrtc screen capture chrome extension example. #87
- JSHint on easyrtc_int.js #169 
- Firefox and previous WebRTC impl does not call emitOnStreamClosed #173 
- getUserMediaConstraints() bug #176 
- Possible bad userAgent RegExp in Adapter.js #177
- A mistake in documentation #188
- wrong condition expression in buildLocalMediaStream #189
- missing argument for wasAcceptedCB #190
- too many arguments passed into onRemoveStreamHelper #192 
- using undefined error code SIGNAL_ERROR #193
- using undefined error code NOVIABLEICE #193
- Possible undefined getUserMedia #194
- Adapter.js Update break easyrtc require-js and AMD support #195 

Reviewed and Merged from other: 
- Delete non existing room fix #84
- Updated /etc/init/easyrtc.conf #101
- Added a bower.json #109
- remove duplicate line #117
- Fixed css class for mirroring local video #153	
- fix more trivia: joinRoom()'s roomParameters not a string #168
- fix unused argument in #200
- fix bug for in #201
- fix Invalid range condition in getIthCaller #202
- fix Unnecessary if branch and flow in InitMediaSource #203 
- fix undefined socket.io method #209

Some already merged or closed, but still for reference.
- WebRTC Version Detecting Fails When Simulating iPad In Chrome Dev Tools #90
- callFailureCB is undefined #133
- FileTransfer support on IE10 #137
- FileTransfer support on IE10 #155
 
cc @dryswabbie
reporters @omochi, @ShiYw, @rretter

hthetiot commented on api/easyrtc.js:
```diff
@@ -3305,39 +3285,43 @@ var Easyrtc = function() {
-                if (mediaIds.hasOwnProperty(streamName) &&
-                        mediaIds[streamName] === webrtcStreamId) {
-                    return streamName;
+            if (self.roomData.hasOwnProperty(roomName)) {
```
Just aded hasOwnProperty, indent cause large diff

hthetiot commented on api/easyrtc_int.js:
```diff
@@ -4666,28 +4720,26 @@ var Easyrtc = function() {
         var added = {}, deleted = {};
         var subPart;
         for (i in newVersion) {
-            if (!newVersion.hasOwnProperty(i)) {
```
15 years of javascript never saw "!newVersion.hasOwnProperty(i)", make my eyes cry.

hthetiot commented on api/easyrtc.js:
```diff
@@ -6310,8 +6318,8 @@ var Easyrtc = function() {
-                     socketError.toString());
+               errorCallback( self.errCodes.SYSTEM_ERROR, 
+                     socketErr.toString());
                return;
```
So annoying to get the real error, an socketError undefined instead of socketErr into to catch.

hthetiot commented on api/easyrtc.js:
```diff
@@ -6084,7 +6092,7 @@ var Easyrtc = function() {
             function iceAddSuccess() {}
             function iceAddFailure(domError) {
-                easyrtc.showError(self.errCodes.ICECANDIDATE_ERR, "bad ice candidate (" + domError.name + "): " +
+                self.showError(self.errCodes.ICECANDIDATE_ERR, "bad ice candidate (" + domError.name + "): " + 
```
why do self at the root and not use it. Did that everywhere.

hthetiot commented on api/easyrtc.js:
```diff
@@ -5217,7 +5215,7 @@ var Easyrtc = function() {
                 message = "Unable to create PeerConnection object, check your ice configuration(" +
-                        JSON.stringify(ice_config) + ")";
+                        JSON.stringify(iceConfig) + ")";
                 if (self.debugPrinter) {
```
Annoying.

hthetiot commented on api/easyrtc.js:
```diff
@@ -4829,7 +4821,7 @@ var Easyrtc = function() {
         //
         if (offersPending[otherUser]) {
-            wasAcceptedCB(true);
+            wasAcceptedCB(true, otherUser);
```
Missing params, annoying.

hthetiot commented on api/easyrtc.js:
```diff
@@ -1955,11 +1929,11 @@ var Easyrtc = function() {
     function event(eventName, callingFunction) {
         if (typeof eventName !== 'string') {
-            self.showError(self.errCodes.DEVELOPER_ERR, src + " called without a string as the first argument");
+            self.showError(self.errCodes.DEVELOPER_ERR, callingFunction + " called without a string as the first argument");
```
Again an annoying undefined.

hthetiot commented on .jshintrc:
```diff
@@ -0,0 +1,122 @@
+{
```
And you are lucky! I could have use JSLINT and not JSHINT.

omochi commented on api/easyrtc.js:
```diff
@@ -2040,7 +2014,8 @@ var Easyrtc = function() {
-        ICECANDIDATE_ERR: "ICECANDIDATE_ERROR"
+        ICECANDIDATE_ERR: "ICECANDIDATE_ERROR",
+        NOVIABLEICE: "NOVIABLEICE_ERROR",
     };
```
It should be following for consistency with others.

```
ICECANDIDATE_ERR: "ICECANDIDATE_ERR",
NOVIABLEICE: "NOVIABLEICE",
```

omochi commented:
I think this patch is nice and great work. (except a tiny one I noted)
@EricDavies please accept these commits and clear issues.

hthetiot commented:
@omochi,  I fixed your review and also applied SIGNAL_ERROR -> SIGNAL_ERR to be the same suffix on every errors "const" - https://github.com/hthetiot/easyrtc/commit/07f550805385c5bb69cccb6450b3464da0ba5830

omochi commented:
@hthetiot thank you 😀

EricDavies commented:
Hi Harold,

I've gotten authorization from my boss to give someone from the easyrtc
community administrative access to the easyrtc github site so that they
apply the minor fixes and pull requests that are needed to bring easyrtc
back to a state of usability. Rod and I just snowed under with work on a
critical project  right now and can't do it. Would you be willing to take
on that chore, at least for a few weeks?

Eric.

On Thu, Feb 18, 2016 at 7:35 PM, Harold Thétiot <notifications@github.com>
wrote:

> It's time to merge or fork.
>
> @EricDavies <https://github.com/EricDavies>,
> Here is a full beta branch that merge with no conflicts and and suppose to
> solve following issues:
>
>    - Add getSourceList and getAudioSourceList #80
>    <https://github.com/priologic/easyrtc/issues/80>
>    - easyrtc screen capture chrome extension example. #87
>    <https://github.com/priologic/easyrtc/issues/87>
>    - JSHint on easyrtc_int.js #169
>    <https://github.com/priologic/easyrtc/issues/169>
>    - Firefox and previous WebRTC impl does not call emitOnStreamClosed
>    #173 <https://github.com/priologic/easyrtc/issues/173>
>    - getUserMediaConstraints() bug #176
>    <https://github.com/priologic/easyrtc/issues/176>
>    - Possible bad userAgent RegExp in Adapter.js #177
>    <https://github.com/priologic/easyrtc/issues/177>
>    - A mistake in documentation #188
>    <https://github.com/priologic/easyrtc/issues/188>
>    - wrong condition expression in buildLocalMediaStream #189
>    <https://github.com/priologic/easyrtc/issues/189>
>    - missing argument for wasAcceptedCB #190
>    <https://github.com/priologic/easyrtc/issues/190>
>    - too many arguments passed into onRemoveStreamHelper #192
>    <https://github.com/priologic/easyrtc/issues/192>
>    - using undefined error code SIGNAL_ERROR #193
>    <https://github.com/priologic/easyrtc/issues/193>
>    - using undefined error code NOVIABLEICE #193
>    <https://github.com/priologic/easyrtc/issues/193>
>    - Possible undefined getUserMedia #194
>    <https://github.com/priologic/easyrtc/issues/194>
>    - Adapter.js Update break easyrtc require-js and AMD support #195
>    <https://github.com/priologic/easyrtc/issues/195>
>
> Some already merged or closed, but still for reference.
>
>    - WebRTC Version Detecting Fails When Simulating iPad In Chrome Dev
>    Tools #90 <https://github.com/priologic/easyrtc/issues/90>
>    - callFailureCB is undefined #133
>    <https://github.com/priologic/easyrtc/issues/133>
>    - FileTransfer support on IE10 #137
>    <https://github.com/priologic/easyrtc/issues/137>
>    - FileTransfer support on IE10 #155
>    <https://github.com/priologic/easyrtc/issues/155>
>
> cc @dryswabbie <https://github.com/dryswabbie>
> reporters @omochi <https://github.com/omochi>, @ShiYw
> <https://github.com/ShiYw>, @rretter <https://github.com/rretter>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/priologic/easyrtc/pull/198
> Commit Summary
>
>    - implement getSourceList and getAudioSourceList #80
>    - fix review #80
>    - add getMediaDevices support #80
>    - merge #60 to beta
>    - easyrtc screen share example #87
>    - fix typo and doc
>    - fix typo and doc for easyrtc_sc #87
>    - merge beta changes
>    - fix ua chrome version parsing #90
>    - Merge branch 'beta2' into beta
>    - merge easyrtc/beta
>    - merge easyrtc beta
>    - fix possible openwebrtc error
>    - merge beta branch
>    - fix AcceptChecker noWebrtcSupport error (and trim space) #133
>    - fix FileTransfer support for IE #137
>    - fix FileTransfer support for IE #155
>    - fix case where MediaStream.hasOwnProperty return fasle on chrome and
>    FF
>    - merge priologic/beta
>    - add .jshintrc config #169
>    - JSHint pass on easyrtc_int.js, no logic change #169
>    - rebuild easyrtc.js #169
>    - merge sylaps/beta
>    - merge beta from priologic
>    - Fix Firefox Does not call emitOnStreamClosed #173
>    - repack easyrtc.js #173
>    - improve isStreamActive for better cross WebRTC impl support #173
>    - repack easyrtc.js #173
>    - fix possible bad chrome userAgent regexp
>    - repack with fixed webrtcDetectedVersion regExp
>    - fix possible bad userAgent RegExp in adapter.js #177
>    - fix Possible undefined getUserMedia #194
>    - fix missing errCodes.NOVIABLEICE #193
>    - fix missing wasAcceptedCB callback argument #190
>    - remove extra argument on onRemoveStreamHelper #192
>    - fix condition expression in buildLocalMediaStream #189
>    - update doc #188
>    - fix getUserMediaConstraints() desiredVideoProperties.height bug #176
>    - merge @hthetiot/beta
>    - merge priologic beta
>    - merge priologic beta
>    - fix bad merge
>    - fix easyrtc AMD support #195
>    - merge
>    - revert regression on #194 after Adapter update
>    - Merge branch 'beta' into beta_sylaps
>    - fix window.getUserMedia usage and use getUserMedia from
>    closure/script #195
>
> File Changes
>
>    - *A* .jshintrc
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-0> (122)
>    - *M* api/adapter.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-1> (47)
>    - *M* api/easyrtc.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-2> (277)
>    - *M* api/easyrtc_ft.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-3> (4)
>    - *M* api/easyrtc_int.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-4> (229)
>    - *A* api/labs/easyrtc_sc.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-5> (255)
>    - *A* api/labs/easyrtc_sc_extension/Icon-16.png
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-6> (0)
>    - *A* api/labs/easyrtc_sc_extension/Icon-64.png
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-7> (0)
>    - *A* api/labs/easyrtc_sc_extension/background-script.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-8> (35)
>    - *A* api/labs/easyrtc_sc_extension/content-script.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-9> (43)
>    - *A* api/labs/easyrtc_sc_extension/manifest.json
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-10> (29)
>    - *M* docs/easyrtc_server_configuration.md
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-11> (7)
>    - *M* docs/easyrtc_server_install.md
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-12> (4)
>    - *M* lib/easyrtc_default_event_listeners.js
>    <https://github.com/priologic/easyrtc/pull/198/files#diff-13> (2)
>
> Patch Links:
>
>    - https://github.com/priologic/easyrtc/pull/198.patch
>    - https://github.com/priologic/easyrtc/pull/198.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/priologic/easyrtc/pull/198>.
>

hthetiot commented:
Yes @EricDavies I will work with @omochi to merge the issues in beta and prepare the next release for you to merge in master if that works for you.

hthetiot commented:
Sincerely thanks @EricDavies

hthetiot commented:
@omochi Ready on my side for one last review.

omochi commented:
@hthetiot I will check later.

hthetiot commented:
@omochi Roger that! I used it without issue for now.

omochi commented:
@hthetiot 

I checked commits following.

> merge Delete non existing room fix #84

Code is ok.

But, I am not sure where this new behavior is good or bad.

When we something delete,
if it is already not exists,
we can think as we get expected result that this is not exists.

I think that consistency is most important thing such case.
So if other deletion apis for entity other than room
returns failure when it is not exists, 
this patch is ok.

> merge Updated /etc/init/easyrtc.conf #101

ok

> merge Added a bower.json #109

ok

> merge remove duplicate line #117

ok

> merge Fixed css class for mirroring local video #153

ok

> fix undefined socket.io method #209

ok. thank you.

> fix Unnecessary if branch and flow in InitMediaSource #203

ok. thank you.

> fix Invalid range condition in getIthCaller #202	

ok. thank you.

> fix bug for in #201

ok. thank you.

> fix unused argument in #200

ok.

But there is regression about `fix bug for in #201`.
please fix it.

> wrong property access of message object for type 'error' in onChannel…
…Cmd #199

ok. thank you.

> fix more trivia: joinRoom()'s roomParameters not a string #168

ok.

> merge New feature: IceConnectionStateChangeListener #122

ok.

> remove SetVideoBandWidth from doc and source after deprecation for mu…
…ltiple release #120

ok.

> error "attempt to connect when already connected to socket server" even if connection failed #98

I wrote my opinion at #98 thread.
Please read it.

> fix Strange code about refreshPane #204	

ok. thank you.

> merge updated example server #112

ok.

hthetiot commented:
Will check you review ASAP

On Saturday, March 5, 2016, omochimetaru <notifications@github.com> wrote:

> @hthetiot <https://github.com/hthetiot>
>
> I checked commits following.
>
> merge Delete non existing room fix #84
> <https://github.com/priologic/easyrtc/pull/84>
>
> Code is ok.
>
> But, I am not sure where this new behavior is good or bad.
>
> When we something delete,
> if it is already not exists,
> we can think as we get expected result that this is not exists.
>
> I think that consistency is most important thing such case.
> So if other deletion apis for entity other than room
> returns failure when it is not exists,
> this patch is ok.
>
> merge Updated /etc/init/easyrtc.conf #101
> <https://github.com/priologic/easyrtc/pull/101>
>
> ok
>
> merge Added a bower.json #109
> <https://github.com/priologic/easyrtc/pull/109>
>
> ok
>
> merge remove duplicate line #117
> <https://github.com/priologic/easyrtc/pull/117>
>
> ok
>
> merge Fixed css class for mirroring local video #153
> <https://github.com/priologic/easyrtc/pull/153>
>
> ok
>
> fix undefined socket.io method #209
> <https://github.com/priologic/easyrtc/issues/209>
>
> ok. thank you.
>
> fix Unnecessary if branch and flow in InitMediaSource #203
> <https://github.com/priologic/easyrtc/issues/203>
>
> ok. thank you.
>
> fix Invalid range condition in getIthCaller #202
> <https://github.com/priologic/easyrtc/issues/202>
>
> ok. thank you.
>
> fix bug for in #201 <https://github.com/priologic/easyrtc/issues/201>
>
> ok. thank you.
>
> fix unused argument in #200
> <https://github.com/priologic/easyrtc/issues/200>
>
> ok.
>
> But there is regression about fix bug for in #201.
> please fix it.
>
> wrong property access of message object for type 'error' in onChannel…
> …Cmd #199 <https://github.com/priologic/easyrtc/issues/199>
>
> ok. thank you.
>
> fix more trivia: joinRoom()'s roomParameters not a string #168
> <https://github.com/priologic/easyrtc/issues/168>
>
> ok.
>
> merge New feature: IceConnectionStateChangeListener #122
> <https://github.com/priologic/easyrtc/issues/122>
>
> ok.
>
> remove SetVideoBandWidth from doc and source after deprecation for mu…
> …ltiple release #120 <https://github.com/priologic/easyrtc/issues/120>
>
> ok.
>
> error "attempt to connect when already connected to socket server" even if
> connection failed #98 <https://github.com/priologic/easyrtc/issues/98>
>
> I wrote my opinion at #98 <https://github.com/priologic/easyrtc/issues/98>
> thread.
> Please read it.
>
> fix Strange code about refreshPane #204
> <https://github.com/priologic/easyrtc/issues/204>
>
> ok. thank you.
>
> merge updated example server #112
> <https://github.com/priologic/easyrtc/pull/112>
>
> ok.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/priologic/easyrtc/pull/198#issuecomment-192626630>.
>


-- 
Harold Thétiot.

hthetiot commented:
@omochi 

About #84 
> But, I am not sure where this new behavior is good or bad.
Will double check

About #201 and #200 i fixed regression

I merged the Jslint 190f7d2 that is only large because i changed the order of "private" function and "expression" for not to be use before being declared.

hthetiot commented:
Ready to merge, will do change log and wait last approval from @omochi and @EricDavies or any other before merging.

omochi commented:
- About #84
Thank you checking. so ok.

- About #201 and #202 
I checked commit `fix regression on #201 from #200	a28ddbc`. ok.

- About Jslint big pass
Yes I know. ok.

- commit `	fix review #198	e119fed`
I checked it. ok.

--

I think that we can merge it now at last !
